programs/clearing_house/src/controller/amm.rs

```diff
@@ -300,6 +306,111 @@ fn calculate_base_swap_output_with_spread(
     ))
 }
 
+pub fn update_spreads(amm: &mut AMM, mark_price: u128) -> ClearingHouseResult<(u128, u128)> {
+    let mut long_spread = (amm.base_spread / 2) as u128;
+    let mut short_spread = (amm.base_spread / 2) as u128;
+
+    if amm.curve_update_intensity > 0 {
+        // oracle retreat
+        // if mark - oracle < 0 (mark below oracle) and user going long then increase spread
+        // msg!("amm.last_oracle_mark_spread_pct: {:?}", amm.last_oracle_mark_spread_pct);
+        if amm.last_oracle_mark_spread_pct < 0 {
+            long_spread = max(long_spread, amm.last_oracle_mark_spread_pct.unsigned_abs());
+        } else {
+            short_spread = max(short_spread, amm.last_oracle_mark_spread_pct.unsigned_abs());
+        }
+
+        // inventory scale
+        let max_invetory_skew = 5 * MARK_PRICE_PRECISION;
+        if amm.total_fee_minus_distributions > 0 {
+            //     let net_cost_basis = cast_to_i128(amm.quote_asset_amount_long)?
+            //         .checked_sub(cast_to_i128(amm.quote_asset_amount_short)?)
+            //         .ok_or_else(math_error!())?;
+
+            let net_base_asset_value = cast_to_i128(amm.quote_asset_reserve)?
```

Comment:
is this the same as doing `_calculate_base_asset_value` on the net base asset position? Do you know which one is more efficient?

Reply:
yes, since there is already terminal_quote_reserve saved, doesnt need to calc new_quote_asset_reserve for the net_base amount